email_account_validity/servlets.py

```diff
@@ -77,11 +79,23 @@ async def _async_render_GET(self, request):
 
         renewal_token = request.args[b"token"][0].decode("utf-8")
 
+        user_id = None
+        if not UNAUTHENTICATED_TOKEN_REGEX.match(renewal_token):
```

Comment:
feels like the decision as to whether authentication is required should be done in the renewal code, rather than here.

So: try to `get_user_by_req` here, but continue if it fails, and then check in `renew_account` or somewhere that there is a `user_id` if one is needed.